### PR TITLE
Add price per item to Quick Order List

### PR DESCRIPTION
**Pull request by sofiamatulis**
### PR Summary:
Add variant list price change to the variant List section when there is volume pricing

### Approach
I am re-using the price per item from Vol Pricing. I had to make some changes to ensure it works for variant list

### Demo links
[Editor](https://os2-demo.myshopify.com/admin/themes/140238880790/editor)

### CODE REVIEW

File: snippets/variant-list-row.liquid [hunk 1]
```diff
@@ -299,3 +364,17 @@
     {% # theme-check-enable %}
   </td>
 </tr>
+
+<volume-pricing class="hidden" id="Volume-{{ variant.id }}">
```
Comment (sofiamatulis) [inline snippets/variant-list-row.liquid:368]: To get the volume pricing, I had to create a hidden volume pricing element to access it

But maybe we can look into an API later

File: snippets/variant-list-row.liquid [hunk 2]
```diff
@@ -77,13 +80,75 @@
           {{ 'products.product.price.sale_price' | t }}
         </dt>
         <dd class="price">
-          <span>{{ 'sections.variant_list.each' | t: money: item_price }}</span>
+          {%- if product.quantity_price_breaks_configured? -%}
+            {%- liquid
+              assign volume_pricing_array = variant.quantity_price_breaks | sort: 'price'
+            -%}
+            <price-per-item
```
Comment (sofiamatulis) [inline snippets/variant-list-row.liquid:87]: Thoughts on making this a snippet?

File: assets/component-volume-pricing.css [hunk 3]
```diff
@@ -41,14 +41,14 @@ volume-pricing show-more-button {
...
   align-items: center;
   gap: 1rem;
 }
 
-.price-per-item {
+.product-form__input .price-per-item {
   font-size: 1.4rem;
   letter-spacing: 0.06rem;
-}
+}
```
Comment (KaichenWang) [inline assets/component-volume-pricing.css:54]:
```suggestion
}

```

File: assets/price-per-item.js [hunk 4]
```diff
@@ -71,10 +75,14 @@ if (!customElements.get('price-per-item')) {
...
+        if (this.classList.contains('variant-item__price-per-item')) {
+          this.currentQtyForVolumePricing = this.getCartQuantity(updatedCartQuantity);
+        }
+
         for (let pair of this.qtyPricePairs) {
           if (this.currentQtyForVolumePricing >= pair[0]) {
-            const pricePerItemCurrent = document.querySelector(`price-per-item[id^="Price-Per-Item-${this.dataset.sectionId}"] .price-per-item`);
-            pricePerItemCurrent.innerHTML = pair[1];
+            const pricePerItemCurrent = document.querySelector(`price-per-item[id^="Price-Per-Item-${this.dataset.sectionId || this.dataset.variantId}"] .price-per-item span`);
+            pricePerItemCurrent.innerHTML = window.variantListStrings.each.replace('[money]', pair[1])
```
Comment (KaichenWang) [inline assets/price-per-item.js:85]: There's an issue with the string displayed on the page after quantity is updated in the Quick order list.

Screenshot 2023-07-07 at 12 52 58 PM png2

File: assets/price-per-item.js [hunk 5]
```diff
@@ -4,8 +4,8 @@ if (!customElements.get('price-per-item')) {
     class PricePerItem extends HTMLElement {
       constructor() {
         super();
-        this.input = document.getElementById(`Quantity-${this.dataset.sectionId}`);
         this.variantId = this.dataset.variantId;
+        this.input = document.getElementById(`Quantity-${this.dataset.sectionId || this.dataset.variantId}`);
```
Comment (KaichenWang) [inline assets/price-per-item.js:8]: `this.input` is null in some cases:

Screenshot 2023-07-07 at 2 05 03 PM

**Review by KaichenWang — COMMENTED** [review]
I'm still testing, but noticed this so far.

File: snippets/variant-list-row.liquid (hunk 2 above)
Comment (eugenekasimov) [reply snippets/variant-list-row.liquid:87]: Eventually we will do it but maybe in a quieter time?

**Review by eugenekasimov — APPROVED** [review]
Works good.

**Comment by KaichenWang** [comment]
Don't think this was introduced in this PR, but adding to cart and then adjusting quantity in Quick order list causes an error. I checked and it seems on cart update, the HTML in the DOM for Quick order errors is replaced with the same element, but with an incorrect id. Quickest fix was just to move the element in Liquid - https://github.com/Shopify/dawn/pull/2798

https://github.com/Shopify/dawn/assets/3710548/ed309194-a615-42fc-aee6-7f64857425e6

**Pull request merged**